detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/SwallowedExceptionSpec.kt

```
@@ -37,6 +37,42 @@ class SwallowedExceptionSpec : Spek({
             assertThat(subject.compileAndLint(code)).hasSize(2)
         }
 
+        it("reports swallowed exceptions only using exception strings via variables") {
+            val code = """
+                fun f() {
+                    try {
+                    } catch (e: IllegalStateException) {
+                        val message = e.message
+                        throw IllegalArgumentException(message)
+                    } catch (f: Exception) {
+                        val message = f.toString()
+                        throw Exception(IllegalArgumentException(message))
+                    }
+                }
+            """
+            assertThat(subject.compileAndLint(code)).hasSize(2)
+        }
+
+        it("reports swallowed exceptions only using exception strings via variables in 'if' block") {
+            val code = """
+                fun f() {
+                    try {
+                    } catch (e: IllegalStateException) {
+                        if (true) {
+                            val message = e.message
+                            throw IllegalArgumentException(message)
+                        }
```

What happens when the user has a construct like this? Is it reported? The exception only gets swallowed when the `condition` is true in that case.

```kotlin
catch (f: Exception) {
  if (condition) {
    val message = f.toString()
    throw Exception(IllegalArgumentException(message))
  }
  throw IllegalStateException(f)
}
```

---

An additional test for this case might be worthwhile.